packages/inter-protocol/src/auction/sortedOffers.js

```diff
@@ -0,0 +1,144 @@
+import { makeRatio } from '@agoric/zoe/src/contractSupport/index.js';
+import { TimeMath } from '@agoric/time';
+
+import { decodeNumber, encodeNumber } from '../vaultFactory/storeUtils.js';
+
+// We want earlier times to sort the same direction as higher prices, so we
+// subtract the timestamp from millisecond time in the year 2286. This works for
+// timestamps in seconds or millis. The alternative considered was inverting,
+// but floats don't have enough resolution to convert back to the same timestamp
+// This will work just fine for at least 250 years. And notice that these
+// timestamps are used for sorting during an auction and don't need to be stored
+// long-term. We could safely subtract from a timestamp that's now + 1 month.
+const FarFuture = 10000000000000n;
+const encodeTimestamp = t => FarFuture - t;
+
+/**
+ * Prices might be more or less than one.
+ *
+ * @param {Ratio} price price quote in IST/Collateral
+ * @returns {number}
+ */
+const priceAsFloat = price => {
+  const n = Number(price.numerator.value);
+  const d = Number(price.denominator.value);
+  return n / d;
+};
+
+/**
+ * Prices might be more or less than one.
+ *
+ * @param {Ratio} discount price quote in IST/IST
+ * @returns {number}
+ */
+const rateAsFloat = discount => {
+  const n = Number(discount.numerator.value);
+  const d = Number(discount.denominator.value);
+  return n / d;
+};
+
+export const toPriceComparator = offerPrice => {
```

Comment:
This name confused me because "comparator" is function, such as given to `sort`.

This string for the keypattern M.gte() https://github.com/Agoric/agoric-sdk/blob/a4017fc3940f8228dcf99ba965ea74b9f4f0435d/packages/store/src/types.js#L576-L577

So maybe `toComparisonOperand` ?